persistence/mongodb-record/src/main/scala/net/liftweb/mongodb/record/field/JObjectField.scala

```diff
@@ -62,21 +52,44 @@ with MongoFieldFlavor[JObject] {
   }
 
   // assume string is json
-  def setFromString(in: String): Box[JObject] = {
+  override def setFromString(in: String): Box[JObject] = {
     // use lift-json to parse string into a JObject
     setBox(tryo(JsonParser.parse(in).asInstanceOf[JObject]))
   }
 
-  def toForm: Box[NodeSeq] = Empty
+  override def toForm: Box[NodeSeq] = Empty
 
-  def asDBObject: DBObject = valueBox
+  override def asDBObject: DBObject = valueBox
     .map { v => JObjectParser.parse(v)(owner.meta.formats) }
     .openOr(new BasicDBObject)
 
-  def setFromDBObject(obj: DBObject): Box[JObject] =
+  override def setFromDBObject(obj: DBObject): Box[JObject] =
     Full(JObjectParser.serialize(obj)(owner.meta.formats).asInstanceOf[JObject])
 
   def setFromDocument(obj: Document): Box[JObject] =
     Full(JObjectParser.serialize(obj)(owner.meta.formats).asInstanceOf[JObject])
 
+  override def asJValue: JValue = valueBox openOr (JNothing: JValue)
+}
+
+class JObjectField[OwnerType <: BsonRecord[OwnerType]](override val owner: OwnerType)
+  extends JObjectTypedField[OwnerType] with MandatoryTypedField[JObject] {
+
+  def this(owner: OwnerType, value: JObject) = {
+    this(owner)
+    setBox(Full(value))
+  }
+
+  override def defaultValue = JObject(List())
+
 }
+
+class OptionalJObjectField[OwnerType <: BsonRecord[OwnerType]](override val owner: OwnerType)
+  extends JObjectTypedField[OwnerType] with OptionalTypedField[JObject] {
+
+  def this(owner: OwnerType, value: Box[JObject]) = {
+    this(owner)
+    setBox(value)
+  }
+
+}
```

Comment:
Please ensure all files end with a newline so that we can be good Unix citizens. :)